client/src/services/dialog/dialog.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { Dialog } from '@interfaces';
+import { Observable, Subject } from 'rxjs';
+import { DialogComponent } from './dialog.component';
+@Injectable({
+  providedIn: 'root',
+})
+export class DialogService {
+  private _dialog$ = new Subject<Dialog | undefined>();
+  public dialogData: Dialog = {
+    title: '',
+    message: '',
+    recipeID: '',
+    primaryBtnText: '',
+    secondaryBtnText: '',
+  };
+
+  constructor(private dialog: MatDialog) {}
+
+  public get dialog$() {
+    return this._dialog$.asObservable();
+  }
+
+  public confirmDelete(recipeID: string): Observable<boolean> {
```

Comment:
This service needs to be generic as well. Rename the method to `open`, similar to `MatDialog`. Then construct dialog data in the recipe id input component and pass it here.